tests/test_enterprise/api/test_views.py

```diff
@@ -627,20 +614,175 @@ def test_enterprise_customer_courses(
 
         assert response == expected
 
+    @ddt.data(
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, False,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, True,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,  # Is a staff user, so it's allowed.
+            'no catalog for query so use empty string, i.e. get everything',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, True,
+            '',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,
+            'no catalog for query so use empty string, i.e. get everything',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, True,
+            'edX Demonstration Course',
+            {
+                'count': 2,
+                'next': None,
+                'previous': None,
+                'results': [
+                    fake_catalog_api.FAKE_SEARCH_ALL_COURSE_RESULT,
+                    fake_catalog_api.FAKE_SEARCH_ALL_SHORT_COURSE_RESULT,
+                ],
+            },
+        ),
+        (
+            PROGRAMS_DETAIL_ENDPOINT,
+            True, True,
+            'uses programs uuid from query parameters for search for programs endpoint',
+            {
+                "count": 1,
+                "next": None,
+                "previous": None,
+                "results": [fake_catalog_api.FAKE_SEARCH_ALL_PROGRAM_RESULT],
+            },
+        ),
+    )
+    @ddt.unpack
+    def test_enterprise_customer_catalogs_detail(self, url, has_linked_user, has_existing_catalog, querystring,
+                                                 expected_search_results):
+        """
+        Make sure the Enterprise Customer's Catalog view returns correctly detailed data.
```

Comment:
Since you wrote it I'll take it, why not :P But normally these details would be explained in the actual code, I'd imagine, and the test's documentation should (usually) be the name of the test function itself, and usually that's plenty.